pkg/cli/scan.go

```diff
@@ -127,8 +160,103 @@ func cmdScan() *cobra.Command {
 	return cmd
 }
 
-func scanInput(inputFilePath string, p *scanParams) (*inputScan, error) {
-	if inputFilePath == "-" {
+func scanInput(inputFile *os.File, p *scanParams) (*inputScan, error) {
+	inputFileName := inputFile.Name()
+
+	// Get the SBOM of the APK
+	var apkSBOM io.Reader
+	if p.sbomInput {
+		apkSBOM = inputFile
+	} else {
+		var s *sbomSyft.SBOM
+		var err error
+
+		if p.disableSBOMCache {
+			s, err = sbom.Generate(inputFileName, inputFile, p.distro)
+		} else {
+			s, err = sbom.CachedGenerate(inputFileName, inputFile, p.distro)
+		}
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate SBOM: %w", err)
+		}
+
+		reader, err := sbom.ToSyftJSON(s)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert SBOM to Syft JSON: %w", err)
+		}
+		apkSBOM = reader
+	}
+
+	// Do the vulnerability scan based on the SBOM
+	result, err := scan.APKSBOM(apkSBOM, p.localDBFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to scan APK using %q: %w", inputFileName, err)
+	}
+
+	is := &inputScan{
+		InputFile: inputFileName,
+		Result:    result,
+	}
+
+	return is, nil
+}
+
+// resolveInputFilePathsFromBuildLog takes the given path to a Melange build log
+// file (or a directory that contains the build log as a "packages.log" file).
+// Once it finds the build log, it parses it, and returns a slice of file paths
+// to APKs to be scanned. Each APK path is created with the assumption that the
+// APKs are located at "./packages/$ARCH/$PACKAGE-$VERSION.apk".
+func resolveInputFilePathsFromBuildLog(buildLogPath string) ([]string, error) {
```

Comment:
🤔 Would it be better to create these APK file paths relative to `filepath.Dir(buildLogPath)`, rather than always relative to `.`?

Reply:
We can follow up if this turns out to be a counterintuitive behavior as-is